cpu/stm32/periph/eth.c

```diff
@@ -626,11 +631,22 @@ static int stm32_eth_confirm_send(netdev_t *netdev, void *info)
             }
             _reset_eth_dma();
         }
-        tx_curr = tx_curr->desc_next;
         tx_bytes += tx_curr->control;
         if (status & TX_DESC_STAT_LS) {
+#ifdef MODULE_NETDEV_TX_INFO_TIMESTAMP
+            if (IS_USED(MODULE_PERIPH_PTP)) {
+                /* time stamp was requested, but check if it actually was captured via TTSS bit */
+                if (status & TX_DESC_STAT_TTSS) {
+                    info->timestamp = tx_curr->ts_low;
+                    info->timestamp += (uint64_t)tx_curr->ts_high * NS_PER_SEC;
+                    info->flags |= NETDEV_TX_INFO_FLAG_TIMESTAMP;
+                }
+            }
+#endif
+            tx_curr = tx_curr->desc_next;
             break;
         }
+        tx_curr = tx_curr->desc_next;
```

Comment:
Is this supposed to advance `tx_curr` twice?

Reply:
It does not. Prior to this PR it was advanced before checking whether to `break` out of the loop, stepping one last time before `break`ing out of the loop (hence, `tx_cur` was pointing one after the last chunk).

Now, this advancing is moved later and is done in either of two cases: The last chunk was found (it is advanced just before `break`) or not the last chunk.

I do believe that the `tx_bytes` was bogus before and is now fixed as side effect, though :facepalm: